### ISSUE TITLE
no-unused-variable false positive with symbol from external package

### ISSUE THREAD

**Issue by donaldpipowitch** [palantir/tslint#3463]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__: (pick one) Node.js API

#### TypeScript code being linted

```ts
import { FormValue } from 'some-component';

type Foo = string; // flagged as unused

export type Baz = FormValue<{
  foo: Foo;
}>;
```

With `some-component` being basically:

```ts
export declare class FormValue<T> {
    value: T;
}
```

#### Actual behavior

```ts
'Foo' is declared but never used. (no-unused-variable)
   7 | import { FormValue } from 'some-component';
   8 |
>  9 | type Foo = string; // flagged as unused
     |     ^
  10 |
  11 | export type Baz = FormValue<{
  12 |   foo: Foo;
```

#### Expected behavior

No error, because `Foo` is used.

---

⚠️  It looks like this problem only happens when `FormValue` is imported from a different package. If I inline the declaration or import it with a relative file path I don't get an error.

**Comment by donaldpipowitch** [comment]
Maybe related? https://github.com/palantir/tslint/issues/3307

**Comment by ajafff** [comment]
Can you try upgrading to typescript@2.6? That seems to have fixed #3307

**Comment by kuhnroyal** [comment]
I am on 2.6.1 and seeing this/https://github.com/palantir/tslint/issues/3484.

**Comment by stphane** [comment]
Such a message may suggest that an (unhandled) syntax/parse error lives somewhere in your code.  
I faced this issue when "Implements" keyword was missing an _interface_ name in front of it.
Ex:  `export class CClientProfilePage implements {`

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
